File: unity/Assets/Scripts/Multiplayer/Players/Ban/Ban.cs
using Michitai.Multiplayer;

namespace Michitai.Multiplayer.Players.Ban
{
    /// <summary>
    /// Static class for ban-related operations
    /// </summary>
    public static class Ban
    {
        /// <summary>
        /// Check if an API response indicates the player is banned
        /// </summary>
        /// <param name="response">The API response</param>
        /// <returns>True if the error indicates the player is banned</returns>
        public static bool IsBanned(ApiResponse response)
        {
            if (!response.success && !string.IsNullOrEmpty(response.error))
            {
                return response.error.Contains("You are banned");
            }
            return false;
        }
    }
}

File: python/multiplayer/players/ban.py
"""Utilities for ban-related operations."""

from __future__ import annotations

from ..api_response import ApiResponse


def is_banned(response: ApiResponse) -> bool:
    """Return True if the API response indicates the player is banned."""
    if not response.success and response.error:
        return "You are banned" in response.error or response.ban_info is not None
    return False
